main/src/library/RedBlackTree.flix

```diff
@@ -78,6 +78,15 @@ namespace RedBlackTree {
             mapAWithKey((_, v) -> v, t)
     }
 
+    instance FunctorFilter[RedBlackTree[k]] with Order[k] {
+        pub def filterMap(f: a -> Option[b] & ef, t: RedBlackTree[k, a]): RedBlackTree[k, b] & ef =
```

Comment:
"Fixed" this, however it's revealed what I think is a bug in typeclass resolution. I've had to implement them as non public functions with different names to the typeclass methods (the "API functions" added to RedBlackTree call these implementations also).

It seems with the enum for RedBlackTree being in the namespace along with the instances then `filter` and `filterMap` did not get resolved to the concrete implementations (the typeclass tests went into an infinite loop). This doesn't seem to be a problem for `Foreach` - presumably because `FilterFunctor` has a default definition and override for `filter` then this is where the cycle is introduced.